### PR TITLE
[Data] Upgrade Arrow version to 15 in CI

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

The most recent stable version of Arrow is 15. So, this PR bumps the version in CI from 14 to 15.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .buildkite/data.rayci.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ steps:
   - name: data6build
     wanda: ci/docker/data6.build.wanda.yaml
 
-  - name: data14build
+  - name: data15build
     wanda: ci/docker/data14.build.wanda.yaml
```
Comment (can-anyscale) [inline .buildkite/data.rayci.yml:11]: change here to data15.build.wanda.yaml

Comment (bveeramani) [reply .buildkite/data.rayci.yml]: Ah, thanks 🤦. Fixed.

**Review by can-anyscale — APPROVED** [review]

File: .buildkite/data.rayci.yml [hunk 2]
```diff
@@ -49,9 +49,9 @@ steps:
       - bazel run //ci/ray_ci:test_in_docker -- //python/ray/data/... //python/ray/air/... data
         --workers "$${BUILDKITE_PARALLEL_JOB_COUNT}" 
         --worker-id "$${BUILDKITE_PARALLEL_JOB}" --parallelism-per-worker 3
-        --build-name data14build
```
Comment (can-anyscale) [inline .buildkite/data.rayci.yml:52]: nit: change the label of this step as well

Comment (bveeramani) [reply .buildkite/data.rayci.yml:52]: Ah, good catch. Thank you!

**Review by aslonnie — APPROVED** [review]

**Pull request merged**